enginetest/memory_server_engine_test.go

```diff
@@ -0,0 +1,769 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enginetest_test
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/dolthub/sqllogictest/go/logictest"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/dolthub/go-mysql-server/enginetest"
+	"github.com/dolthub/go-mysql-server/enginetest/queries"
+	"github.com/dolthub/go-mysql-server/enginetest/scriptgen/setup"
+	memharness "github.com/dolthub/go-mysql-server/enginetest/sqllogictest/harness"
+	"github.com/dolthub/go-mysql-server/memory"
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+type ServerEngineTestSuite struct {
+	suite.Suite
+	hmu           *sync.Mutex
+	memoryHarness *enginetest.MemoryHarness
+}
+
+// this function executes before each test case
+func (suite *ServerEngineTestSuite) SetupTest() {
+	suite.setHarness(enginetest.NewDefaultMemoryHarness())
+}
+
+// this function executes after each test case
+func (suite *ServerEngineTestSuite) TearDownTest() {
+	suite.setHarness(nil)
+}
+
+// setHarness is called from any Test that uses non-default MemoryHarness.
+// It sets the suite harness to given harness and calls UseServer() method of it.
+func (suite *ServerEngineTestSuite) setHarness(mh *enginetest.MemoryHarness) {
+	suite.hmu.Lock()
+	defer suite.hmu.Unlock()
+	suite.memoryHarness = mh
+	if mh != nil {
+		suite.memoryHarness.UseServer()
+	}
+}
+
+func TestServerEngineTestSuite(t *testing.T) {
+	if _, ok := os.LookupEnv("CI_TEST"); ok {
+		t.Skip("Skipping ServerEngineTestSuite for CI_TEST=true for now")
+	}
+	s := new(ServerEngineTestSuite)
+	s.hmu = &sync.Mutex{}
```

Comment:
Do we need the mutex locking? Are multiple threads calling `setHarness`? I thought the test runner would run tests in serial from a single thread. If that's true, I think you can avoid the mutex here.